File: java/ComplexServer.java

import org.apache.thrift.server.TServer;
import org.apache.thrift.server.TServer.Args;
import org.apache.thrift.server.TSimpleServer;
import org.apache.thrift.transport.TServerSocket;
import org.apache.thrift.transport.TServerTransport;

public class ComplexServer {
    public static ComplexCalculatorHandler handler;
    public static CalculatorComplex.Processor processor;

    public static void main(String [] args) {
        try {
            handler = new ComplexCalculatorHandler();
            processor = new CalculatorComplex.Processor(handler);

            Runnable simple = new Runnable() {
                public void run() {
                    simple(processor);
                }
            };

            new Thread(simple).start();
        } catch (Exception x) {
            x.printStackTrace();
        }
    }

    public static void simple(CalculatorComplex.Processor processor) {
        try {
            TServerTransport serverTransport = new TServerSocket(9090);
            TServer server = new TSimpleServer(new Args(serverTransport).processor(processor));

            System.out.println("Starting the complex server...");
            server.serve();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

}

File: python/complex_server.py
import glob
import sys

from calculator import CalculatorComplex
from calculator import ttypes

from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol
from thrift.server import TServer

import logging

logging.basicConfig(level=logging.DEBUG)


class CalculadoraHandler:
    def __init__(self):
        self.log = {}

    def add(self, operands):
        if ( len(operands.complexOperand1) != len(operands.complexOperand2) ):
            raise ttypes.InvalidOperation("Los tamaños deben de ser iguales")
        result = []
        for i in range(0, len(operands.complexOperand1)):
            result.append(operands.complexOperand1[i] + operands.complexOperand2[i])
        return result

    def multiply(self, operands):
        if ( len(operands.complexOperand1) != len(operands.complexOperand2) ):
            raise ttypes.InvalidOperation("Los tamaños deben de ser iguales")
        return operands.complexOperand1

    def subtract(self, operands):
        if ( len(operands.complexOperand1) != len(operands.complexOperand2) ):
            raise ttypes.InvalidOperation("Los tamaños deben de ser iguales")
        result = []
        for i in range(0, len(operands.complexOperand1)):
            result.append(operands.complexOperand1[i] - operands.complexOperand2[i])
        return result


if __name__ == "__main__":
    handler = CalculadoraHandler()
    processor = CalculatorComplex.Processor(handler)
    transport = TSocket.TServerSocket(host="127.0.0.1", port=9090)
    tfactory = TTransport.TBufferedTransportFactory()
    pfactory = TBinaryProtocol.TBinaryProtocolFactory()

    server = TServer.TSimpleServer(processor, transport, tfactory, pfactory)

    print("iniciando servidor complejo...")
    server.serve()
    print("fin")
